Orange/widgets/utils/state_summary.py

```diff
@@ -0,0 +1,74 @@
+from Orange.data import StringVariable, DiscreteVariable, ContinuousVariable, \
+    TimeVariable
+
+
+def format_variables_string(variables):
+    """
+    A function that formats the descriptive part of the input/output summary for
+    either features, targets or metas of the input dataset.
+
+    :param variables: Features, targets or metas of the input dataset
+    :return: A formatted string
+    """
+    agg = []
+    for var_type_name, var_type in [('categorical', DiscreteVariable),
+                                    ('numeric', ContinuousVariable),
+                                    ('time', TimeVariable),
+                                    ('string', StringVariable)]:
+        # Disable pylint here because a `TimeVariable` is also a
+        # `ContinuousVariable`, and should be labelled as such. That is why
+        # it is necessary to check the type this way instead of using
+        # `isinstance`, which would fail in the above case
+        var_type_list = [v for v in variables if type(v) is var_type]  # pylint: disable=unidiomatic-typecheck
+        if var_type_list:
+            shown = var_type in (StringVariable,)
+            agg.append(
+                (f'{len(var_type_list)} ' + var_type_name +
+                 f"{['', ' (not shown)'][shown]}",
+                 len(var_type_list)))
+
+    if not agg:
+        return '—'
+
+    attrs, counts = list(zip(*agg))
+    if len(attrs) > 1:
+        var_string = ', '.join(attrs[:-1]) + ', ' + attrs[-1]
+        return f'{sum(counts)} (' + var_string + ')'
+    elif sum(counts) == 1:
+        var_string = attrs[0][2:]
```

Comment:
I had problems understanding this code (before I realized that the first two characters are always `"1 "`. Please add a comment `# remove leading "1 ". :)